### DISCUSSION TITLE
When the Create 3 CPU 100% utilization issue will be solved ?

### DISCUSSION THREAD

**Discussion by sashag1991** [iRobotEducation/create3_docs#403] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.1.2

### Which RMW is your robot running?

FastRTPS

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I have TB4 that use create 3 platform. I see from TB4 forums that this platform have issue of 100% utilization in CPU.
It avoids me to continue my work with TB4. The robot have delays in data packets and also it harms the navigation due to the fact it cant avoid obstacles

**Comment by mdxtinkernick** [comment]
there is no mention of this in the bug fix list for just released firmware update, so afraid it may still be an issue :(

**Comment by sashag1991** [comment]
Is there a workaround to overcome the issue? change to other ROS2 version or something else ?
I can't proceed with my tasks because the robot cant navigate properly, it not avoids obstacles.

**Comment by shamlian** [comment]
TurtleBot configuration is not technically in scope for this repository. (The Create 3 robot on its own does _not_ normally consume 100% CPU.) However, the iRobot team is working to reduce CPU utilization in general on the Create 3 robot. We have just released [G.5.2](https://iroboteducation.github.io/create3_docs/releases/g_5_2/) and [H.2.2](https://iroboteducation.github.io/create3_docs/releases/h_2_2/), which reduce core loading by about 9%. Please give the appropriate version for your release of ROS 2 a try (I assume Humble if you're on TB4). I'd suggest following https://github.com/iRobotEducation/create3_docs/issues/381 for updates.
